### DISCUSSION TITLE
Hide Copilot Button

### DISCUSSION THREAD

**Discussion by ecnepsnai** [community/community#141033] [Mobile]
How do I hide that annoying always present copilot button?

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.180.0 (142788027)
Device: iPhone 15 Pro Max (iOS 18.0.1)
</details>

**Comment by stevepopovich** [comment]
Hi @ecnepsnai you should be able to hide the button, if not you are experiencing a bug.

If you don't have a Copilot subscription, tap the button and dismiss the banner. Then the button will be gone.

If you have a Copilot subscription, you can always disable the button in Profile -> Settings -> Copilot.

**Reply by meneguzzi** [reply]
I don't have a subscription, and these co-pilot buttons show up everywhere in my organisation and private personal repositories. I don't want to see Co-Pilot anywhere near my code. If this isn't removed, I'm very inclined to move away from Github altogether.

**Reply by KalleOlaviNiemitalo** [reply]
I had **two** Copilot buttons in GitHub Mobile version 1.183.1 (145298821) on iOS 18.1.  Both of them were near the bottom right corner of the screen, overlapping so that the upper button was almost completely hidden behind the lower button; only a small part of the goggles in the Copilot icon was visible, and it looked like a connected ⁀⁀ symbol, or upside-down ⏖ symbol.  Rotating the screen from portrait to landscape or back did not change this.

I touched the lower button and dismissed the advertising banner; there was a pop-up message referring to Copilot in settings, but the buttons stayed there.  I did the same with the mostly-occluded upper button (zooming in to make sure I was hitting the visible part of it), to the same effect.  Opened the settings menu and chose Copilot from there but didn't find a way to remove the buttons.  Then cleared the cache, terminated the GitHub Mobile app, and started it again; and finally the buttons stopped appearing.

**Reply by stevepopovich** [reply]
@KalleOlaviNiemitalo you saw 2 buttons? Do you have a screenshot of that?

**Reply by KalleOlaviNiemitalo** [reply]
@stevepopovich, no I don't have a screenshot.  If the problem recurs, I'll get one for sure.

I originally thought they were only one button and the ⁀⁀ was some kind of expand-up symbol.  However, when I touched the lower button, only the lower button shrank a little to indicate the touch; the upper button stayed where it was.

**Reply by KalleOlaviNiemitalo** [reply]
@stevepopovich, does the GitHub mobile app save the banner dismissal into client-side data only, or into my user profile in GitHub servers?  In other words, if I were to purge and reinstall the app, would the Copilot button reappear so that I could then check whether it's doubled?

**Reply by stevepopovich** [reply]
That setting is client-side, so a reinstall will reset the button, yes 👍🏼

**Reply by KalleOlaviNiemitalo** [reply]
Got some screenshots.

Portrait orientation, one Copilot button some distance above the bottom bar:

![image](https://github.com/user-attachments/assets/c2c48c42-e9c7-4a2d-b59c-093147631d05)

Landscape orientation, one Copilot button some distance above the bottom bar:

![image](https://github.com/user-attachments/assets/5be462d3-fedf-4615-8ec2-b3986bed4277)

Portrait orientation, one Copilot button overlapping the bottom bar, i.e. lower than in the first screenshot:

![image](https://github.com/user-attachments/assets/303bd2f8-ffc8-4c37-921f-9729180a3121)

Portrait orientation, two Copilot buttons, where the lower button is in front of the upper button and also overlaps the bottom bar:

![image](https://github.com/user-attachments/assets/1cbe9706-fdef-4724-af1a-bf7cd76b5493)

Landscape orientation, two Copilot buttons, where the lower button is in front of the upper button but neither button overlaps the bottom bar:

![image](https://github.com/user-attachments/assets/542a6657-536d-43ec-9479-6f8b4e4741bd)

Seems related to flipping between landscape and portrait orientation.

**Reply by KalleOlaviNiemitalo** [reply]
@stevepopovich, I'm also using Lockdown mode in iOS; do you think it makes a difference here?

**Reply by stevepopovich** [reply]
wow! Thanks for capturing this. Yeah, I will file this as a bug and we will look into fixing it. Thank you very much!

**Reply by KalleOlaviNiemitalo** [reply]
# Repro in more detail

1. Uninstall and reinstall the GitHub Mobile app.
2. From the home screen, start the app in portrait orientation and sign in with a GitHub account that does not have Copilot.  The Copilot button does not appear yet. 
3. Double-press the home button of the phone to get the app switcher, and slide the app up to close it.
4. Start the app again from the home screen.  A Copilot button appears near the bottom right corner.
5. Press the home button once to return from the app to the home screen.  From there, touch the GitHub Mobile app icon to return to the app.

   (At this point, there are already two Copilot buttons, but they are at the exact same position so only one is visible.  If you carefully touched and held the button in the front, then it would shrink a little and reveal the borders of the button in the back. But don’t do that yet, because it’d risk showing the Copilot banner.)
6. Turn the phone from portrait to landscape orientation.  This moves the buttons so that they don’t completely overlap, and you can see that there are two.

**Reply by stevepopovich** [reply]
Thank you for the incredible detail!

**Reply by KalleOlaviNiemitalo** [reply]
When I repeatedly switch to the GitHub Mobile app by touching its icon on the home screen, the shadow around the Copilot button gradually becomes darker.  I think this means the app adds another button each time and that multiplies the shadow effect.  Here's a screenshot after touching the icon 30 times:

![image](https://github.com/user-attachments/assets/b7082457-c841-4465-a33b-414aace423fd)

At the same time, scrolling also becomes a little less smooth, presumably because of the time spent rendering the buttons and shadows.

**Reply by KalleOlaviNiemitalo** [reply]
The version history for GitHub version 1.189.0 for iOS, released on December 11, 2024, claims:

> \- Fixed Copilot button duplication.

I have not tested this fix yet.

**Comment by cybardev** [comment]
With the latest update, you can't anymore...

**Reply by escape0707** [reply]
Absolute Microslop behavior

**Reply by cybardev** [reply]
I particularly hate it cuz it replaced the much useful Profile button. Now I need to tap Home, then extend my thumb to the top-left corner to tap my profile pic to reach the same screen. Useless slop indeed...

**Reply by KalleOlaviNiemitalo** [reply]
Thanks for the warning!  Now I'll know not to update.

**Reply by lucascantor** [reply]
I guess it's time to delete the app then!

**Reply by PushkarajDevulkar** [reply]
Microsoft Copilot is good guys! Its helpful especially for unlimited messages with no limits even in free tier! Why you guys so hate you're showing to it. Its good for natural language conversation, yet it makes mistakes but its unlimited so enjoy, because you're getting unlimited messages' ability in Copilot's free tier with dedicated GPT 5 for smart work.

**Reply by ardrigh** [reply]
Copilot being forced upon Github users because Microsoft is burning billions of dollars so they are faking user engagement metrics this way instead.

I don't care if there is a "Free" I still don't want it.
